src/Elastic.Apm/Model/Transaction.cs

```diff
@@ -120,54 +122,100 @@ internal Transaction(ApmAgent agent, string name, string type)
 					_activity.TraceId.CopyTo(idBytesFromActivity);
 					// Read right most bits. From W3C TraceContext: "it is important for trace-id to carry "uniqueness" and "randomness" in the right part of the trace-id..."
 					idBytesFromActivity = idBytesFromActivity.Slice(8);
+
+					_traceState = new TraceState();
+
+					// If activity has a tracestate, populate the transaction tracestate with it.
+					if (!string.IsNullOrEmpty(_activity.TraceStateString))
+						_traceState.AddTextHeader(_activity.TraceStateString);
+
 					IsSampled = sampler.DecideIfToSample(idBytesFromActivity.ToArray());
+
+					// In the unlikely event that tracestate populated from activity contains an es vendor key, the tracestate
+					// is mutated to set the sample rate defined by the sampler, because we consider a transaction without
+					// explicitly passed distributedTracingData to be a **root** transaction. The end result
+					// is that activity tracestate will be propagated, along with the sample rate defined by this transaction.
+					if (IsSampled)
+					{
+						SampleRate = sampler.Rate;
+						_traceState.SetSampleRate(sampler.Rate);
+					}
+					else
+					{
+						SampleRate = 0;
+						_traceState.SetSampleRate(0);
+					}
+
+					// sync the activity tracestate with the tracestate of the transaction
+					_activity.TraceStateString = _traceState.ToTextHeader();
 				}
 				else
 				{
-					// In case from some reason the activity creation was not successful, let's create new random ids
+					// If an activity wasn't created, generate values
 					var idBytes = new byte[8];
 					Id = RandomGenerator.GenerateRandomBytesAsString(idBytes);
 					IsSampled = sampler.DecideIfToSample(idBytes);
 
 					idBytes = new byte[16];
 					TraceId = RandomGenerator.GenerateRandomBytesAsString(idBytes);
+
+					if (IsSampled)
+					{
+						_traceState = new TraceState(sampler.Rate);
+						SampleRate = sampler.Rate;
+					}
+					else
+					{
+						_traceState = new TraceState(0);
+						SampleRate = 0;
+					}
 				}
 
-				// PrentId could be also set here, but currently in the UI each trace must start with a transaction where the ParentId is null,
+				// ParentId could be also set here, but currently in the UI each trace must start with a transaction where the ParentId is null,
 				// so to avoid https://github.com/elastic/apm-agent-dotnet/issues/883 we don't set it yet.
 			}
 			else
 			{
 				if (_activity != null)
+				{
 					Id = _activity.SpanId.ToHexString();
+					// TODO: should _activity's trace context be populated with values from distributedTracingData?
```

Comment:
Good question - we have never done that and so far there was no feedback on this. Having said that, if we don't do it, then in this case the data between Activity and our transaction will be different. So I'm leaning toward syncing the data here as well.

Reply:
This has now been done as part of #1228, so I'll remove the comment 

https://github.com/elastic/apm-agent-dotnet/blob/eb4f85b79b2b7b942d60a54c7a7b1256ca3d4343/src/Elastic.Apm/Model/Transaction.cs#L148-L155